test/unit/core/observer.ts

```diff
@@ -522,5 +535,22 @@ suite('Observer', ({ expect, spy, stub }) => {
 
       expect(emit).to.be.calledWith(`${Events.UI_UPDATED}:${tagName}:${id}`, OBJ[tagName][id]);
     });
+    it('should emit nothing if states same', () => {
+      observers.ui(false, false);
+
+      expect(emit.notCalled).to.be.true;
+    });
+    it('should not emit UI_UPDATED event if oldTagState and newTagState same', () => {
+      const tagName = 'Main';
+      const id = 'brand';
+      const OBJ = {
```

Comment:
this object can be all one line,
also local constants should be camelcase, not uppercase

Reply:
Is `testObject` an acceptable name?